### PR TITLE
Procedures for spatial matches

### PR DESCRIPTION
**Pull request by mgraber**
#223

### CODE REVIEW

File: knownprojects_build/_sql/_procedures.sql [hunk 1]
```diff
@@ -0,0 +1,168 @@
+/*
```
Comment (SPTKL) [inline knownprojects_build/_sql/_procedures.sql:1]: the concern here is that if we create this procedure at this step, `_combined` doesn't exist yet, will a new postgres database throw an error for creating a procedure on a table that doesn't exist?

Comment (mgraber) [reply knownprojects_build/_sql/_procedures.sql:1]: Good point. We can move them either to the script where they get called, or to a procedure declaration script that gets called later in the process.

File: knownprojects_build/_sql/correct_projects.sql [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+	
+	-- Remove first record_id from array, then loop through the remaining record_ids	
+	record_id_array = array_remove(record_id_array, _first_record_id);
+	
+	<<reassign_remaining>>
+	FOREACH _record_id IN ARRAY record_id_array LOOP
+	    -- Add record_id to cluster containing record_id_match
+		CALL reassign_single_record(_record_id, _first_record_id);
+	END LOOP reassign_remaining;
+
```
Comment (SPTKL) [inline knownprojects_build/_sql/correct_projects.sql:81]: this is really cool!

**Review by SPTKL — APPROVED** [review]

File: knownprojects_build/_sql/correct_projects.sql [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+BEGIN
+	SELECT record_id_array[1] INTO _first_record_id;
+	
+	-- Move first record_id, either to existing project or new project
+	CALL reassign_single_record(_first_record_id, record_id_match);
+	
+	-- Remove first record_id from array, then loop through the remaining record_ids	
+	record_id_array = array_remove(record_id_array, _first_record_id);
+	
+	<<reassign_remaining>>
```
Comment (SPTKL) [inline knownprojects_build/_sql/correct_projects.sql:76]: will the `<<>>` notation throw errors?

**Pull request merged**